### PR TITLE
Defrag datalink/v3

### PR DESCRIPTION
**Pull request by victorjulien**
https://redmine.openinfosecfoundation.org/issues/6887
https://redmine.openinfosecfoundation.org/issues/6943

SV https://github.com/OISF/suricata-verify/pull/1756

### CODE REVIEW

**Comment by victorjulien** [comment]
Replaces #10807.

File: src/decode.h [hunk 1]
```diff
@@ -995,6 +995,7 @@ void DecodeUnregisterCounters(void);
  * Libpcap on at least OpenBSD returns 101 as datalink type for RAW pcaps though. */
 #define LINKTYPE_RAW2        101
 #define LINKTYPE_IPV4        228
+#define LINKTYPE_IPV6        229
```
Comment (catenacyber) [inline src/decode.h:998]: Where did you find this ? Wireshark code ? Some IANA registration ?

**Review by catenacyber — APPROVED** [review]
Thanks for the work Victor :-)

This PR looks good to me, but I do not approve the SV one (as it would turn CI red for other branches)

- CI : 🟢 
- Code : ok
- Commits segmentation : ok
- Commit messages : nice
- Git ID set : looks fine for me
- CLA : you already contributed :-p
- Doc update : not needed
- Redmine ticket : ok, but I do not think these deserve a backport to 6
- Rustfmt : not needed
- Tests : 🟠 SV PR needs min-version: 8
- Dependencies added: none

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 20024

File: src/defrag.c [hunk 2]
```diff
@@ -861,6 +863,9 @@ DefragInsertFrag(ThreadVars *tv, DecodeThreadVars *dtv, DefragTracker *tracker,
 #ifdef DEBUG
     new->pcap_cnt = pcap_cnt;
 #endif
+    if (frag_offset == 0) {
+        tracker->datalink = p->datalink;
```
[the remark this answers is not in the document]
Comment (victorjulien) [reply src/defrag.c:867]: Yes, tunnel packets get DLT_RAW as they lack the L2 layers.

File: src/decode.h (hunk 1 above)
Comment (victorjulien) [reply src/decode.h:998]:
```
/usr/include/pcap/dlt.h:#define DLT_IPV6		229
```
btw a follow up task could be to do string representation of the datalink. We know of at least one OS difference:
```
/* taken from pcap's bpf.h */
  #ifndef DLT_RAW
  #ifdef __OpenBSD__
  #define DLT_RAW     14  /* raw IP */
  #else
  #define DLT_RAW     12  /* raw IP */
  #endif
  #endif
  ```

**Comment by victorjulien** [comment]
Merged in #10832, thanks!

**Pull request closed without merging**